modules/ingester/instance.go

```diff
@@ -113,26 +114,32 @@ func (i *instance) PushBytes(ctx context.Context, id tempodb_encoding.ID, object
 
 // Moves any complete traces out of the map to complete traces
 func (i *instance) CutCompleteTraces(cutoff time.Duration, immediate bool) error {
+	now := time.Now()
+	tracesToCut := list.New()
```

Comment:
what about using a slice here, which would avoid the boxing of values into interface{} and let you use a range statement on line 132?

Reply:
Agree with this, we can estimate slice capacity using `len(i.traces)` to avoid using `append()`

Thanks for the review @davecheney :)

Reply:
Unsure about this.  Each time CutCompleteTraces is called only a very small percentage of the traces in the map will actually end up in the list/slice.  It would be difficult to estimate the length of the slice.

Going to try some benchmarks.

Reply:
Predictably when a low number of traces is cut from the active traces map the linked list implementation performs better.  That advantage is slowly lost as more and more traces are cut.

```
$ go test -bench=BenchmarkCutTraces* -benchmem
goos: linux
goarch: amd64
pkg: github.com/grafana/tempo/modules/ingester
BenchmarkCutTraces1-8            	    5420	    229893 ns/op	     105 B/op	       2 allocs/op
BenchmarkCutTraces10-8           	    3825	    276236 ns/op	     549 B/op	      11 allocs/op
BenchmarkCutTraces100-8          	    3654	    288851 ns/op	    4871 B/op	     101 allocs/op
BenchmarkCutTraces1000-8         	    3024	    370372 ns/op	   48070 B/op	    1001 allocs/op
BenchmarkCutTraces5000-8         	    1792	    771957 ns/op	  240075 B/op	    5001 allocs/op
BenchmarkCutTraces10000-8        	    1376	   1003460 ns/op	  480084 B/op	   10001 allocs/op
BenchmarkCutTracesSlice1-8       	    3433	    319624 ns/op	   81934 B/op	       1 allocs/op
BenchmarkCutTracesSlice10-8      	    3636	    322590 ns/op	   81933 B/op	       1 allocs/op
BenchmarkCutTracesSlice100-8     	    3628	    309431 ns/op	   81933 B/op	       1 allocs/op
BenchmarkCutTracesSlice1000-8    	    3114	    326222 ns/op	   81936 B/op	       1 allocs/op
BenchmarkCutTracesSlice5000-8    	    3135	    345734 ns/op	   81935 B/op	       1 allocs/op
BenchmarkCutTracesSlice10000-8   	    3338	    328694 ns/op	   81934 B/op	       1 allocs/op
PASS
```

However, after reviewing the code I realized I am wrong above.  The _normal_ situation is for most traces to be cut from the active map.  Switching to a slice implementation.  Thanks for the review!